Minio/V4Authenticator.cs

```diff
@@ -166,25 +184,52 @@ private string GetSignedHeaders(SortedDictionary<string, string> headersToSign)
             return string.Join(";", headersToSign.Keys);
         }
 
+        /// <summary>
+        /// Determines and returns the kind of service
+        /// </summary>
+        /// <param name="assumeRole">boolean; if true role credentials, otherwise IAM user</param>
+        /// <returns>returns the kind of service as a string</returns>
+        private string getServiceKind(bool assumeRole)
+        {
+            var serviceKind = "s3";
+            if (assumeRole) serviceKind = "sts";
+            return serviceKind;
+        }
+
```

Comment:
This is also an incorrect API `assumeRole` is a single API it should be called `getService(bool sts)`

Reply:
OK. I'll modify it.